Tests/SonarScanner.MSBuild.TFS.Test/Classic/BinaryToXmlCoverageReportConverterTests.cs

```diff
@@ -81,52 +82,17 @@ public void Conv_OutputIsCaptured()
             var inputFilePath = Path.Combine(testDir, "input.txt");
             File.WriteAllText(inputFilePath, "dummy input file");
 
-            var converterFilePath = Path.Combine(testDir, "converter.bat");
-            File.WriteAllText(converterFilePath,
-@"
-echo Normal output...
-echo Error output...>&2
-echo Create a new file using the output parameter
-echo foo > """ + outputFilePath + @"""");
-
             // Act
-            var success = BinaryToXmlCoverageReportConverter.ConvertBinaryToXml(converterFilePath, inputFilePath, outputFilePath, logger);
+            var success = BinaryToXmlCoverageReportConverter.ConvertBinaryToXml(inputFilePath, outputFilePath, logger);
 
             // Assert
-            success.Should().BeTrue("Expecting the process to succeed");
+            success.Should().BeFalse();
 
-            File.Exists(outputFilePath).Should().BeTrue("Expecting the output file to exist");
-            TestContext.AddResultFile(outputFilePath);
+            File.Exists(outputFilePath).Should().BeFalse("Conversion failed");
 
-            logger.AssertInfoLogged("Normal output...");
-            logger.AssertErrorLogged("Error output...");
-        }
-
-        [TestMethod]
-        [WorkItem(72)] // Regression test for bug #72: fail the conversion if the output file is not created
-        public void Conv_FailsIfFileNotFound()
```

Comment:
What happens if we provide null or a wrong path?

Reply:
This is a good question. It wasn't under test before. This test handled the case when the tool failed and checked for the output file and we have the equivalent here:

https://github.com/SonarSource/sonar-scanner-msbuild/blob/9fcddb156008d2217bc8ab7adb4238f749656de4/Tests/SonarScanner.MSBuild.TFS.Test/Classic/BinaryToXmlCoverageReportConverterTests.cs#L96

There are some `Debug.Assert` in the implementation about the input, but I think we should turn these into proper exceptions and put them under test.

https://github.com/SonarSource/sonar-scanner-msbuild/blob/9fcddb156008d2217bc8ab7adb4238f749656de4/src/SonarScanner.MSBuild.TFS.Classic/BinaryToXmlCoverageReportConverter.cs#L73-L75

What do you think?

Reply:
Yes please, I think it would be better. 

While doing this, please check that the application will not fail. In my opinion, we should raise a warning about wrong paths but not stop the analysis.